### ISSUE TITLE
Error in development using the preact plugin

### ISSUE THREAD

**Issue by Angelmmiguel** [zeit/next-plugins#191]
Hello!

I'm getting this issue when I access to my application in the development environment:

```
Cannot use 'in' operator to search for 'href' in undefined
TypeError: Cannot use 'in' operator to search for 'href' in undefined
    at ProxyComponent.render (node_modules/next/dist/lib/link.js:168:0)
    at ProxyComponent.hotComponentRender (node_modules/react-hot-loader/dist/react-hot-loader.development.js:579:0)
    at ProxyComponent.proxiedRender (node_modules/react-hot-loader/dist/react-hot-loader.development.js:587:0)
    at callMethod (node_modules/preact-compat/dist/preact-compat.es.js:491:0)
    at ProxyComponent.render (node_modules/preact-compat/dist/preact-compat.es.js:502:0)
    at renderComponent (node_modules/preact/dist/preact.esm.js:728:0)
    at setComponentProps (node_modules/preact/dist/preact.esm.js:676:0)
    at buildComponentFromVNode (node_modules/preact/dist/preact.esm.js:856:0)
    at idiff (node_modules/preact/dist/preact.esm.js:397:0)
    at innerDiffNode (node_modules/preact/dist/preact.esm.js:516:0)
```

At that point, the `child` variable in [next/link](https://github.com/zeit/next.js/blob/canary/lib/link.js#L142) doesn't have the proper `props` property initialized, so the `in` operator fails.

After some investigation, it seems the issue is related with [react-hot-loader](https://github.com/gaearon/react-hot-loader). There are an open issue about component initialization in Preact: https://github.com/developit/preact/issues/1007

@mopduan opened the same issue 1 month ago, but it was closed without any reply: #167. 

Thanks for the help!

**Comment by developit** [comment]
This was an UglifyJS bug, and has been fixed in Preact [8.3.0](https://github.com/developit/preact/releases/tag/8.3.0).

**Comment by Angelmmiguel** [comment]
Thank you for solving it! :)
